highlevel/robot/controller/motion/trajectory.py

```diff
@@ -0,0 +1,71 @@
+"""
+Trajectory controller module.
+"""
+import math
+
+from highlevel.logger import LOGGER
+from highlevel.robot.controller.motion.motion import MotionController
+from highlevel.robot.controller.motion.position import PositionController
+from highlevel.robot.controller.obstacle import ObstacleController
+from highlevel.robot.entity.type import MotionResult
+from highlevel.util.geometry.vector import Vector2
+
+
+class TrajectoryController:
+    """
+    Trajectory controller, allows movements between two points, using a pathfinding algorithm.
+    """
+    def __init__(self, position_controller: PositionController,
+                 motion_controller: MotionController,
+                 obstacle_controller: ObstacleController):
+        self.obstacle_controller = obstacle_controller
+        self.motion_controller = motion_controller
+        self.position_controller = position_controller
+
+    async def move_to(self, pos_target: Vector2, reverse: bool) -> None:
+        """
+        Moves the robot to a new position, avoiding obstacles in the way.
+        """
+
+        pos_current = self.position_controller.position
+
+        LOGGER.get().info('trajectory_controller_move_to',
+                          current=pos_current,
+                          target=pos_target,
+                          reverse=reverse)
+
+        if (pos_current - pos_target).euclidean_norm() < 1:
+            return
+
+        direction = pos_target - pos_current
+        distance = direction.euclidean_norm()
```

Comment:
you can compute distance oncee and move the condition after..

Reply:
did you remove the condition alltogether? why?

Reply:
put it back, as suggested.